### PR TITLE
Ccp 828 more features

### PR DESCRIPTION
**Pull request by chrisEff**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

- [ ] Bug fix
- [ ] New feature
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I updated the CHANGELOG.md

### CODE REVIEW

**Review by philipp-heyse — APPROVED** [review]

File: extension/lib/hooks/getRootCategories-after-addSmartCategories.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+
+module.exports = async (context, input) => {
+  if (!context.config.smartCategories) {
+    return input
```
Comment (sofib) [inline extension/lib/hooks/getRootCategories-after-addSmartCategories.js]: It is nicer if we don't use input as output

Comment (chrisEff) [reply extension/lib/hooks/getRootCategories-after-addSmartCategories.js]: done

**Review by sofib — CHANGES_REQUESTED** [review]
We need to hook into getCategory and getCategoryChildren as well.

File: README.md [hunk 2]
```diff
@@ -12,19 +12,43 @@ Set the following values in your Shopgate Connect Admin:
...
   * `name` - (text, required) Name of the category that will be shown to the end user.
-  * `filters` - (object, required) The filters in CloudSearch syntax.
+  * `searchPhrase` - (text, optional) A searchPhrase to be executed when the user opens this category.
+  * `filters` - (object, optional) The filters in CloudSearch syntax.
+  * `productIds` - (array, optional) List of product IDs.
   * `description` - (text, optional) Not diplayed anywhere atm. Just for documentary purposes.
   * `imageUrl` - (text, optional) Link to the category image.
-  * `externalUrl` - (text, optional)  Link to the category in the desktop shop.
+  * `externalUrl` - (text, optional) Link to the category in the desktop shop.
+  * `hideFromTree` - (true/false, optional, default: false) Don't show the smart category in the category tree.
```
Comment (sofib) [inline README.md]: Please reverse the config to say "showInCategoryTree" and that it defaults to true.

**Review by philipp-heyse — APPROVED** [review]

**Pull request merged**